bundles/pubsub/pubsub_protocol/pubsub_protocol_lib/src/pubsub_wire_protocol_common.c

```diff
@@ -62,8 +62,9 @@ static celix_status_t pubsubProtocol_createNetstring(const char* string, char**
                 return CELIX_ENOMEM;
             }
         }
-        *netstringOutLength = numlen + str_len + 2;
-        sprintf(*netstringOut, "%zu:%s,", str_len, string);
+        *netstringOutLength = numlen + str_len + 2; //Note +2 for ':', ','
+        int written = snprintf(*netstringOut, *netstringOutLength+1, "%zu:%s,", str_len, string); //adding +1 for '\0'
```

Comment:
`if(*netstringOut == NULL)`, it is unclear where `1024 >= numlen+str_len+2+1`.
In `else if (*netstringMemoryOutLength < numlen + str_len + 2)`, it may happen that `*netstringMemoryOutLength == numlen + str_len + 2`, then `snprintf(*netstringOut, *netstringOutLength+1)` may lead to memory overwrite by 1 byte.

Reply:
IMO this function in its current state is too complex. I will try to refactor this using open_memstream instead of snprintf.